partner_contact_lang/models/res_partner.py

```diff
@@ -10,15 +10,13 @@ class ResPartner(models.Model):
 
     @api.multi
     def write(self, vals):
-        if 'lang' in vals and vals['lang']:
-            if 'child_ids' in vals:
-                childs = self.browse(vals['child_ids'])
-            else:
-                childs = self.mapped('child_ids')
-            childs = childs.filtered(lambda x: not x.lang)
+        """Propagate a language change in the partner to the child contacts."""
+        res = super(ResPartner, self).write(vals)
+        if vals.get('lang'):
+            childs = self.mapped('child_ids').filtered(lambda x: not x.lang)
```

Comment:
I think that here you'd better use `self.search([('id','child_of',self.ids),('lang','=',False)])`, because I remember some addon allowing you to have more than 1 level of "childness" (not sure which ATM).